healthcare/api-client/v1/fhir/fhir_stores_test.py

```diff
@@ -0,0 +1,190 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pytest
+import sys
+import time
+import uuid
+
+from google.cloud import exceptions, storage
+
+# Add datasets for bootstrapping datasets for testing
+sys.path.append(os.path.join(os.path.dirname(__file__), "..", "datasets"))  # noqa
+import datasets
+import fhir_stores
+
+cloud_region = "us-central1"
+project_id = os.environ["GOOGLE_CLOUD_PROJECT"]
+service_account_json = os.environ["GOOGLE_APPLICATION_CREDENTIALS"]
+
+dataset_id = "test_dataset_{}".format(uuid.uuid4())
+fhir_store_id = "test_fhir_store-{}".format(uuid.uuid4())
+
+gcs_uri = os.environ["CLOUD_STORAGE_BUCKET"]
+RESOURCES = os.path.join(os.path.dirname(__file__), "resources")
+source_file_name = "Patient.json"
+resource_file = os.path.join(RESOURCES, source_file_name)
+import_object = gcs_uri + "/" + source_file_name
+
+
+@pytest.fixture(scope="module")
+def test_dataset():
+    dataset = datasets.create_dataset(
+        service_account_json, project_id, cloud_region, dataset_id
+    )
+
+    yield dataset
+
+    # Clean up
+    datasets.delete_dataset(service_account_json, project_id, cloud_region, dataset_id)
+
+
+def test_CRUD_fhir_store(test_dataset, capsys):
+    fhir_stores.create_fhir_store(
+        service_account_json, project_id, cloud_region, dataset_id, fhir_store_id
+    )
+
+    fhir_stores.get_fhir_store(
+        service_account_json, project_id, cloud_region, dataset_id, fhir_store_id
+    )
+
+    fhir_stores.list_fhir_stores(
+        service_account_json, project_id, cloud_region, dataset_id
+    )
+
+    fhir_stores.delete_fhir_store(
+        service_account_json, project_id, cloud_region, dataset_id, fhir_store_id
+    )
+
+    out, _ = capsys.readouterr()
+
+    # Check that create/get/list/delete worked
+    assert "Created FHIR store" in out
+    assert "Name" in out
+    assert "fhirStores" in out
+    assert "Deleted FHIR store" in out
+
+
+def test_patch_fhir_store(test_dataset, capsys):
+    fhir_stores.create_fhir_store(
+        service_account_json, project_id, cloud_region, dataset_id, fhir_store_id
+    )
+
+    fhir_stores.patch_fhir_store(
+        service_account_json, project_id, cloud_region, dataset_id, fhir_store_id
+    )
+
+    # Clean up
+    fhir_stores.delete_fhir_store(
+        service_account_json, project_id, cloud_region, dataset_id, fhir_store_id
+    )
+
+    out, _ = capsys.readouterr()
+
+    assert "Patched FHIR store" in out
+
+
+def test_import_fhir_store_gcs(test_dataset, capsys):
+    fhir_stores.create_fhir_store(
+        service_account_json, project_id, cloud_region, dataset_id, fhir_store_id
+    )
+
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(gcs_uri)
+    blob = bucket.blob(source_file_name)
+
+    blob.upload_from_filename(resource_file)
+
+    time.sleep(10)  # Give new blob time to propagate
```

Comment:
This leads to a test flakiness. Consider a more reliable way.

Reply:
Using retrying package to check for HttpError with backoff.